### PR TITLE
feat: Account index validator

### PR DESCRIPTION
**Pull request by maxwellmattryan**
# Description of change

Adds validation for account index in the Ledger migration.

## Links to any relevant issues

None

## Type of change

- Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

Tested on (w/ Ledger Nano S): 

- Ubuntu (20.04)
- MacOS (Catalina 10.15.70

## Change checklist

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: packages/shared/routes/setup/ledger/views/AccountIndex.svelte [hunk 1]
```diff
@@ -9,11 +9,20 @@
     export let locale
     export let mobile
 
+    let expert = false
     let loading = false
 
+    let min = 0
+    let max = 2147483647
```
Comment (laumair) [inline packages/shared/routes/setup/ledger/views/AccountIndex.svelte:16]: @maxwellmattryan Why this number?

**Review by laumair — APPROVED** [review]

File: packages/shared/routes/setup/ledger/views/AccountIndex.svelte (hunk 1 above)
Comment (maxwellmattryan) [reply packages/shared/routes/setup/ledger/views/AccountIndex.svelte:16]: I believe this is what the max index was in Trinity.

File: packages/shared/routes/setup/ledger/views/AccountIndex.svelte [hunk 2]
```diff
@@ -25,6 +34,25 @@
...
+            n = Math.min(Math.max(n, min), max)
+
+        return n
+    }
+
+    function isValidNumber(n: number): boolean {
+        return isPositiveInteger(n) && isWithinRange(n)
+    }
+
+    function isPositiveInteger(n: number): boolean {
```
Comment (begonaalvarezd) [inline packages/shared/routes/setup/ledger/views/AccountIndex.svelte:48]: @maxwellmattryan maybe we could add some error messages for the validators? I pasted for example `194909e` into the input and its not allowing me to continue but there is no feedback.
image


Maybe we could use something like we have in `ImportTextfield.svelte`  

image

Comment (begonaalvarezd) [reply packages/shared/routes/setup/ledger/views/AccountIndex.svelte:48]: If we add these validators, I think we can remove the `validate` function in `packages/shared/components/inputs/Number.svelte` which forces a min/max

**Pull request merged**